OCPBUGS#10545: Configuring cluster wide proxy on the External DNS Operator (#57402)
openshift/openshift-docs: modules/configuring-egress-proxy-edns-operator.adoc

// Module included in the following assemblies:
//
// * networking/external_dns_operator/nw-configuring-cluster-wide-egress-proxy.adoc

:_content-type: PROCEDURE
[id="nw-configuring-cluster-wide-proxy_{context}"]

> = Configuring the cluster-wide proxy on the External DNS Operator

Reviewer: ```suggestion
= Configuring the cluster-wide egress proxy on the External DNS Operator
```

Author: As discussed - https://github.com/openshift/openshift-docs/pull/57402#discussion_r1148746649, opting for using the cluster-wide proxy instead of cluster-wide egress proxy

> You must trust a custom certificate authority (CA) and create the OLM subscription of the External DNS Operator.

Reviewer: ```suggestion
Configuring the External DNS Operator to trust the certificate authority of the cluster-wide egress proxy.
```
`Must trust` is not accurate, the egress proxy may not have the untrusted certificate. Also, to match what we said in ALBO doc.

Author: Addressed - maintained the same format as ALBO. Removed the sentence and updated the heading

. Create the config map to contain the CA bundle in the `external-dns-operator` namespace by running the following command:
+
[source,terminal]
----
$ oc -n external-dns-operator create configmap trusted-ca
----

. To inject the trusted CA bundle into the config map, add the `config.openshift.io/inject-trusted-cabundle=true` label to the config map by running the following command:
+
[source,terminal]
----
$ oc -n external-dns-operator label cm trusted-ca config.openshift.io/inject-trusted-cabundle=true
----

> . Update the subscription of the External DNS Operator to access the config map in the deployment of the External DNS Operator by running the following command:

Reviewer: ```suggestion
. Update the subscription of the External DNS Operator by running the following command:
```

Author: Addressed

+
[source,terminal]
----
$ oc -n external-dns-operator patch subscription external-dns-operator --type='json' -p='[{"op": "add", "path": "/spec/config", "value":{"env":[{"name":"TRUSTED_CA_CONFIGMAP_NAME","value":"trusted-ca"}]}}]'
----

. After the deployment of the External DNS Operator is completed, verify that the trusted CA environment variable is added to the `external-dns-operator` deployment by running the following command:
+
[source,terminal]
----

> $ oc -n external-dns-operator exec deploy/external-dns-operator -c external-dns-operator -- printenv TRUSTED_CA_CONFIGMAP_NAME

Reviewer: Example output would be:
```
trusted-ca
```

// Module included in the following assemblies:
//
// * networking/external_dns_operator/nw-configuring-cluster-wide-egress-proxy.adoc

:_content-type: PROCEDURE
[id="nw-configuring-cluster-wide-proxy_{context}"]
= Configuring the External DNS Operator to trust the certificate authority of the cluster-wide proxy

> . Create the config map to contain the CA bundle in the `external-dns-operator` namespace by running the following command:

Reviewer 2: This missing a module introduction and the `.Procedure` heading.

Author: Addressed

+
[source,terminal]
----
$ oc -n external-dns-operator create configmap trusted-ca
----

. To inject the trusted CA bundle into the config map, add the `config.openshift.io/inject-trusted-cabundle=true` label to the config map by running the following command:
+
[source,terminal]
----
$ oc -n external-dns-operator label cm trusted-ca config.openshift.io/inject-trusted-cabundle=true
----

. Update the subscription of the External DNS Operator by running the following command:
+
[source,terminal]
----
$ oc -n external-dns-operator patch subscription external-dns-operator --type='json' -p='[{"op": "add", "path": "/spec/config", "value":{"env":[{"name":"TRUSTED_CA_CONFIGMAP_NAME","value":"trusted-ca"}]}}]'
----

> . After the deployment of the External DNS Operator is completed, verify that the trusted CA environment variable is added to the `external-dns-operator` deployment by running the following command:

Reviewer 2: Not critical, but this could be added as a step (unnumbered) in a .Verification section.

Author: Addressed
